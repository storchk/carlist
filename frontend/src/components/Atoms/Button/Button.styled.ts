import styled, { css } from 'styled-components'

export const StyledButton = styled.button(
  ({ theme }) => css`
    display: inline-flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacing.sm} ${theme.spacing.lg};
    border-radius: ${theme.border.radius};
    transition: all 0.5s cubic-bezier(0.2, 0, 0.05, 1);
    background-color: ${theme.colors.primary};
    color: ${theme.colors.contrast0};

    &:hover {
      cursor: pointer;
      box-shadow: rgba(0, 0, 0, 0.1) 0px 0.5rem 8px 0px;
      transform: scale(1.02);
    }
  `
)
