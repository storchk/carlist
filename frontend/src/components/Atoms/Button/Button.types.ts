import type { AnchorHTMLAttributes } from 'react'

export type ButtonProps = {
  label: string
} & AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement>
