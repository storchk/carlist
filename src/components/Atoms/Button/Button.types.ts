import type { AnchorHTMLAttributes } from 'react'

export type ButtonProps = {
  isLoading?: boolean
  label: string
} & AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement>
