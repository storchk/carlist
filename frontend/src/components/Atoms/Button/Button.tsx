import type { MouseEvent } from 'react'

import { Typography } from '../Typography'
import { StyledButton } from './Button.styled'
import type { ButtonProps } from './Button.types'

export const Button = ({ label, onClick }: ButtonProps): JSX.Element => {
  const handleOnClick = (event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => {
    event.stopPropagation()

    if (onClick) {
      onClick(event)
    }
  }

  return (
    <StyledButton onClick={handleOnClick}>
      <Typography fontSize={{ mobile: 'sm' }} fontWeight="bold">
        {label}
      </Typography>
    </StyledButton>
  )
}
