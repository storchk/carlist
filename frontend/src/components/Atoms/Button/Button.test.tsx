import { axe } from 'jest-axe'

import { testing } from '@/testing'
import { Button } from './Button'
import type { ButtonProps } from './Button.types'

const { render, screen, renderer } = testing
const label = 'Hello world! I am using React'

const renderComponent = (props?: Partial<ButtonProps>) =>
  render(<Button label={label} {...props} />)
describe('atoms: Button', () => {
  it('should render button with text', () => {
    renderComponent()
    const text = screen.getByText(/Hello world! I am using React/i)
    expect(text).toBeInTheDocument()
  })

  it('snapshot', () => {
    const view = renderer(<Button label={label} />)
    expect(view).toMatchSnapshot()
  })

  it('a11y - No violations detected', async () => {
    const { container } = renderComponent()
    await expect(axe(container)).resolves.toHaveNoViolations()
  })
})
