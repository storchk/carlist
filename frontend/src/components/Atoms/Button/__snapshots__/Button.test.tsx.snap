// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms: Button snapshot 1`] = `
.c1 {
  font-weight: 700;
  color: inherit;
  font-size: 16px;
}

.c0 {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 16px 32px;
  border-radius: 8px;
  transition: all 0.5s cubic-bezier(0.2, 0, 0.05, 1);
  background-color: #f7b628;
  color: #ffffff;
}

.c0:hover {
  cursor: pointer;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0.5rem 8px 0px;
  transform: scale(1.02);
}

<button
  className="c0"
  onClick={[Function]}
>
  <span
    className="c1"
    fontSize={
      {
        "mobile": "sm",
      }
    }
    fontWeight="bold"
  >
    Hello world! I am using React
  </span>
</button>
`;
